src/nu/validator/saxtree/LocatorImpl.java

```diff
@@ -62,11 +68,13 @@ public LocatorImpl(Locator locator) {
             this.publicId = null;
             this.column = -1;
             this.line = -1;
+            this.encoding = null;
         } else {
             this.systemId = locator.getSystemId();
             this.publicId = locator.getPublicId();
             this.column = locator.getColumnNumber();
             this.line = locator.getLineNumber();
+            this.encoding = ((Locator2)locator).getEncoding();
```

Comment:
Please put this behind an `instanceof` check to keep saxtree usable outside this parser.

Reply:
Done